build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```diff
@@ -0,0 +1,78 @@
+import java.nio.charset.StandardCharsets
+import java.util.Optional
+import java.util.jar.JarFile
+import java.util.stream.Stream
+import java.util.stream.Collectors
+import groovy.transform.Field
+
+/**
+ * This Gradle script captures Predictive Test Selection and Test Distribution compatibility for each Test task, 
+ * adding a flag as custom value.
+ * This should be applied to the root project since it configures all projects:
+ * <code> apply from: file('gradle-test-pts-support.gradle') </code>
+ */
+
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+    return
+}
+
+@Field
+final def supportedEngines = [
+    'org.junit.support.testng.engine.TestNGTestEngine' : 'testng',
+    'org.junit.jupiter.engine.JupiterTestEngine' : 'junit-jupiter', 
+    'org.junit.vintage.engine.VintageTestEngine' : 'junit-vintage', 
+    'org.spockframework.runtime.SpockEngine' : 'spock',
+    'net.jqwik.engine.JqwikTestEngine' : 'jqwik',
+    'io.cucumber.junit.platform.engine.CucumberTestEngine' : 'cucumber',
+    'org.spekframework.spek2.junit.SpekTestEngine' : 'spek2',
```

Comment:
Those are not supported https://docs.gradle.com/enterprise/predictive-test-selection/#_frameworks_and_languages

Reply:
Oh, I made this list given the GE smoke tests (testing test discovery for each fwk) assuming it was enough for PTS compatibility.